Playback freezes when seeking back to time with EXT-X-GAPs

### What version of Hls.js are you using?

1.5.17

### What browser (including version) are you using?

Chrome 130.0.6723.70, Firefox 131.0.3

### What OS (including version) are you using?

Windows 11 23H2

### Test stream

https://hlsjs.video-dev.org/demo/?src=https%3A%2F%2Fgist.githubusercontent.com%2Fluryus%2Ff54bc840dbe29e69a07b43ca93d0eec9%2Fraw%2F752c23978cf885a36b1675f85cd42615057c5360%2Fbunny_gaps.m3u8&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsInN0b3BPblN0YWxsIjpmYWxzZSwiZHVtcGZNUDQiOmZhbHNlLCJsZXZlbENhcHBpbmciOi0xLCJsaW1pdE1ldHJpY3MiOi0xfQ==

### Configuration

```JavaScript
{
  "debug": true,
  "enableWorker": true,
  "lowLatencyMode": true,
  "backBufferLength": 90
}
```


### Additional player setup steps

_No response_

### Checklist

- [X] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [X] The issue occurs in the stable client (latest release) on https://hlsjs.video-dev.org/demo and not just on my page
- [X] The issue occurs in the latest client (main branch) on https://hlsjs-dev.video-dev.org/demo and not just on my page
- [X] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [X] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. Start playback of the linked test stream. The stream contains GAP segments from about 0:20 to 1:10
2. Seek to a time after the gap such that the segment right after the gap is no longer buffered
3. Seek back to the gap
4. Observe the playback stall

See this screen capture: 

https://github.com/user-attachments/assets/6ffcd411-dcc8-4172-9da0-ce6136952415



### Expected behaviour

The player skips forward to the segment after the gap, and continues playback.

We're trying to use `EXT-X-GAP`s to "fill" the playlist in case there are short outages in the video source (e.g. a camera reboot). Basically a similar use case to that in issue #6518 . Is this even the right approach here, or should we do something different instead on the server side?

### What actually happened?

The playback stalls when seeking to a time with gaps. This issue happens on both Chrome and Firefox. Safari on macOS works fine.


### Console output

```shell
...
 [log] > [transmuxer.ts]: Flushed fragment 14 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 14 of level 0 (frag:[139.993-150.023] > buffer:[70.023-150.001])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: Loading fragment 15 cc: 0 of [0-63] level: 0, target: 150.001
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Loaded fragment 15 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
 [log] > [transmuxer.ts]: Flushed fragment 15 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 15 of level 0 (frag:[150.001-160.023] > buffer:[70.023-160.009])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: media seeking to 137.566, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 137.566, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 137.566, state: IDLE
 [log] > [stream-controller]: Loading fragment 16 cc: 0 of [0-63] level: 0, target: 160.009
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Loaded fragment 16 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
 [log] > [transmuxer.ts]: Flushed fragment 16 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 16 of level 0 (frag:[160.009-170.023] > buffer:[70.023-169.993])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: Media seeked to 137.566
 [log] > [stream-controller]: media seeking to 150.127, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 150.127, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 150.127, state: IDLE
 [log] > [stream-controller]: Loading fragment 17 cc: 0 of [0-63] level: 0, target: 169.993
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Media seeked to 150.127
 [log] > [stream-controller]: Loaded fragment 17 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
 [log] > [transmuxer.ts]: Flushed fragment 17 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 17 of level 0 (frag:[169.993-180.023] > buffer:[70.023-180.001])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: Loading fragment 18 cc: 0 of [0-63] level: 0, target: 180.001
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Loaded fragment 18 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
 [log] > [transmuxer.ts]: Flushed fragment 18 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 18 of level 0 (frag:[180.001-190.023] > buffer:[70.023-190.009])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: media seeking to 163.884, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 163.884, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 163.884, state: IDLE
 [log] > [stream-controller]: Loading fragment 19 cc: 0 of [0-63] level: 0, target: 190.009
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Loaded fragment 19 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
 [log] > [transmuxer.ts]: Flushed fragment 19 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 19 of level 0 (frag:[190.009-200.023] > buffer:[70.023-199.993])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: Media seeked to 163.884
 [log] > [stream-controller]: media seeking to 174.051, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 174.051, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 174.051, state: IDLE
 [log] > [stream-controller]: Loading fragment 20 cc: 0 of [0-63] level: 0, target: 199.993
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Media seeked to 174.051
 [log] > [buffer-controller] Removing [0,78.56562500000001] from the audio SourceBuffer
 [log] > [buffer-controller] Removing [0,78.56562500000001] from the video SourceBuffer
 [log] > [stream-controller]: Loaded fragment 20 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
 [log] > [transmuxer.ts]: Flushed fragment 20 of level 0
 [log] > [stream-controller]: PARSING->PARSED
 [log] > [stream-controller]: Buffered main sn: 20 of level 0 (frag:[199.993-210.023] > buffer:[79.990-210.001])
 [log] > [stream-controller]: PARSED->IDLE
 [log] > [stream-controller]: media seeking to 35.887, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 35.887, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 35.887, state: IDLE
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 3 cc: 0 of [0-63] level: 0, target: 35.887
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-63] level: 0, target: 40.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 0, target: 50.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 0, target: 60.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
 [warn] > skipping hole, adjusting currentTime from 35.886899 to 80.0399
_trySkipBufferHole @ hls.js:27248
 [log] > [stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 80.040, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [stream-controller]: Media seeked to 80.040
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
 [log] > [stream-controller]: media seeking to 34.691, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 34.691, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 34.691, state: IDLE
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 3 cc: 0 of [0-63] level: 0, target: 34.691
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-63] level: 0, target: 40.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 0, target: 50.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 0, target: 60.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
 [warn] > skipping hole, adjusting currentTime from 34.690669 to 80.0399
_trySkipBufferHole @ hls.js:27248
 [log] > [stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 80.040, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [stream-controller]: Media seeked to 80.040
 [log] > [stream-controller]: media seeking to 40.074, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 40.074, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 40.074, state: IDLE
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-63] level: 0, target: 40.074
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 0, target: 50.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 0, target: 60.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
 [warn] > skipping hole, adjusting currentTime from 40.073704 to 80.0399
_trySkipBufferHole @ hls.js:27248
 [log] > [stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 80.040, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [stream-controller]: Media seeked to 80.040
 [log] > [stream-controller]: media seeking to 54.428, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 54.428, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 54.428, state: IDLE
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 0, target: 54.428
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 0, target: 60.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
 [warn] > skipping hole, adjusting currentTime from 54.428463 to 80.0399
_trySkipBufferHole @ hls.js:27248
 [log] > [stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 80.040, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [stream-controller]: Media seeked to 80.040
 [log] > [stream-controller]: media seeking to 28.710, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 28.710, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 28.710, state: IDLE
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 2 cc: 0 of [0-63] level: 0, target: 28.71
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 3 cc: 0 of [0-63] level: 0, target: 30.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-63] level: 0, target: 40.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 0, target: 50.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 0, target: 60.008
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ hls.js:24928
 [log] > [stream-controller]: FRAG_LOADING->IDLE
 Error event: Object
(anonymous) @ hls-demo.js:24783
 [info] > [abr] buffer is empty, optimal quality level 0
 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
 [warn] > skipping hole, adjusting currentTime from 28.709519 to 80.0399
_trySkipBufferHole @ hls.js:27248
 [log] > [stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 80.040, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 80.040, state: IDLE
 [log] > [stream-controller]: Media seeked to 80.040
 [log] > [stream-controller]: media seeking to 198.574, state: IDLE
 [log] > [audio-stream-controller]: media seeking to 198.574, state: STOPPED
 [log] > [subtitle-stream-controller]: media seeking to 198.574, state: IDLE
 [log] > [stream-controller]: Loading fragment 21 cc: 0 of [0-63] level: 0, target: 210.001
 [log] > [stream-controller]: IDLE->FRAG_LOADING
 [log] > [stream-controller]: Loaded fragment 21 of level 0
 [log] > [stream-controller]: FRAG_LOADING->PARSING
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 21 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->PARSED
level-controller.ts:549 [log] > [level-controller]: Resetting level error count of 18 on frag buffered
base-stream-controller.ts:570 [log] > [stream-controller]: Buffered main sn: 21 of level 0 (frag:[210.001-220.023] > buffer:[79.990-220.009])
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 22 cc: 0 of [0-63] level: 0, target: 220.009
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 22 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 22 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->PARSED
stream-controller.ts:556 [log] > [stream-controller]: Media seeked to 198.575
base-stream-controller.ts:570 [log] > [stream-controller]: Buffered main sn: 22 of level 0 (frag:[220.009-230.023] > buffer:[79.990-229.994])
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSED->IDLE
buffer-controller.ts:1096 [log] > [buffer-controller] Removing [0,108.3125] from the audio SourceBuffer
buffer-controller.ts:1096 [log] > [buffer-controller] Removing [0,108.3125] from the video SourceBuffer
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 23 cc: 0 of [0-63] level: 0, target: 229.994
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 23 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 23 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:570 [log] > [stream-controller]: Buffered main sn: 23 of level 0 (frag:[229.994-240.023] > buffer:[108.357-240.001])
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:241 [log] > [stream-controller]: media seeking to 35.289, state: IDLE
base-stream-controller.ts:241 [log] > [audio-stream-controller]: media seeking to 35.289, state: STOPPED
base-stream-controller.ts:241 [log] > [subtitle-stream-controller]: media seeking to 35.289, state: IDLE
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 3 cc: 0 of [0-63] level: 0, target: 35.289
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
content-steering-controller.ts:206 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ content-steering-controller.ts:206
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->IDLE
main.js:745 Error event: Object
(anonymous) @ main.js:745
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-63] level: 0, target: 40.008
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
content-steering-controller.ts:206 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ content-steering-controller.ts:206
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->IDLE
main.js:745 Error event: Object
(anonymous) @ main.js:745
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 0, target: 50.008
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
content-steering-controller.ts:206 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ content-steering-controller.ts:206
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->IDLE
main.js:745 Error event: Object
(anonymous) @ main.js:745
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 0, target: 60.008
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
content-steering-controller.ts:206 [warn] > Could not resolve fragGap ("GAP tag found") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":461.10000002384186}
onError @ content-steering-controller.ts:206
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->IDLE
main.js:745 Error event: Object
(anonymous) @ main.js:745
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:1155 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:1155 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
abr-controller.ts:607 [info] > [abr] buffer is empty, optimal quality level 0
base-stream-controller.ts:1155 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 7
```


### Chrome media internals output

_No response_

--- a/api-extractor/report/hls.js.api.md
+++ b/api-extractor/report/hls.js.api.md
@@ -1866,7 +1866,7 @@ export class FragmentTracker implements ComponentAPI {
     getBufferedFrag(position: number, levelType: PlaylistLevelType): MediaFragment | null;
     // (undocumented)
     getFragAtPos(position: number, levelType: PlaylistLevelType, buffered?: boolean): MediaFragment | null;
-    getPartialFragment(time: number): Fragment | null;
+    getPartialFragment(time: number): MediaFragment | null;
     // (undocumented)
     getState(fragment: Fragment): FragmentState;
     // (undocumented)

--- a/api-extractor/report/hls.js.api.md
+++ b/api-extractor/report/hls.js.api.md
@@ -2019,6 +2019,8 @@ class Hls implements HlsEventEmitter {
     get loadLevel(): number;
     // Warning: (ae-setter-with-docs) The doc comment for the property "loadLevel" must appear on the getter, not the setter.
     set loadLevel(newLevel: number);
+    // (undocumented)
+    get loadLevelObj(): Level | null;
     loadSource(url: string): void;
     readonly logger: ILogger;
     get lowLatencyMode(): boolean;

--- a/docs/API.md
+++ b/docs/API.md
@@ -190,6 +190,7 @@ See [API Reference](https://hlsjs-dev.video-dev.org/api-docs/) for a complete li
   - [Interstitial Objects and Classes](#interstitial-objects-and-classes)
 - [Additional data](#additional-data)
   - [`hls.latestLevelDetails`](#hlslatestleveldetails)
+  - [`hjs.loadLevelObj`](#hjsloadlevelobj)
   - [`hls.sessionId`](#hlssessionid)
 - [Runtime Events](#runtime-events)
 - [Creating a Custom Loader](#creating-a-custom-loader)
@@ -2145,7 +2146,11 @@ type InterstitialAssetErrorData = {
 
 ### `hls.latestLevelDetails`
 
-- get: Returns the LevelDetails of the most up-to-date HLS variant playlist data.
+- get: Returns the `LevelDetails` of last loaded level (variant) or `null` prior to loading a media playlist.
+
+### `hjs.loadLevelObj`
+
+- get: Returns the `Level` object of the selected level (variant) or `null` prior to selecting a level or once the level is removed.
 
 ### `hls.sessionId`
 

--- a/src/controller/abr-controller.ts
+++ b/src/controller/abr-controller.ts
@@ -672,8 +672,8 @@ class AbrController extends Logger implements AbrComponentAPI {
     }
     // If no matching level found, see if min auto level would be a better option
     const minLevel = hls.levels[minAutoLevel];
-    const autoLevel = hls.levels[hls.loadLevel];
-    if (minLevel?.bitrate < autoLevel?.bitrate) {
+    const autoLevel = hls.loadLevelObj;
+    if (autoLevel && minLevel?.bitrate < autoLevel.bitrate) {
       return minAutoLevel;
     }
     // or if bitrate is not lower, continue to use loadLevel

--- a/src/controller/base-stream-controller.ts
+++ b/src/controller/base-stream-controller.ts
@@ -1215,11 +1215,11 @@ export default class BaseStreamController
         bufferedFragAtPos &&
         (bufferInfo.nextStart <= bufferedFragAtPos.end || bufferedFragAtPos.gap)
       ) {
-        return BufferHelper.bufferInfo(
-          bufferable,
-          pos,
-          Math.max(bufferInfo.nextStart, maxBufferHole),
+        const gapDuration = Math.max(
+          Math.min(bufferInfo.nextStart, bufferedFragAtPos.end) - pos,
+          maxBufferHole,
         );
+        return BufferHelper.bufferInfo(bufferable, pos, gapDuration);
       }
     }
     return bufferInfo;

--- a/src/controller/error-controller.ts
+++ b/src/controller/error-controller.ts
@@ -176,7 +176,7 @@ export default class ErrorController
       case ErrorDetails.SUBTITLE_LOAD_ERROR:
       case ErrorDetails.SUBTITLE_TRACK_LOAD_TIMEOUT:
         if (context) {
-          const level = hls.levels[hls.loadLevel];
+          const level = hls.loadLevelObj;
           if (
             level &&
             ((context.type === PlaylistContextType.AUDIO_TRACK &&
@@ -200,7 +200,7 @@ export default class ErrorController
         return;
       case ErrorDetails.KEY_SYSTEM_STATUS_OUTPUT_RESTRICTED:
         {
-          const level = hls.levels[hls.loadLevel];
+          const level = hls.loadLevelObj;
           const restrictedHdcpLevel = level?.attrs['HDCP-LEVEL'];
           if (restrictedHdcpLevel) {
             data.errorAction = {

--- a/src/controller/fragment-tracker.ts
+++ b/src/controller/fragment-tracker.ts
@@ -321,7 +321,7 @@ export class FragmentTracker implements ComponentAPI {
   /**
    * Gets the partial fragment for a certain time
    */
-  public getPartialFragment(time: number): Fragment | null {
+  public getPartialFragment(time: number): MediaFragment | null {
     let bestFragment: Fragment | null = null;
     let timePadding: number;
     let startTime: number;

--- a/src/controller/gap-controller.ts
+++ b/src/controller/gap-controller.ts
@@ -541,11 +541,11 @@ export default class GapController extends TaskLoop {
                 PlaylistLevelType.MAIN,
               );
             if (startProvisioned) {
-              // Do not seek when switching variants
-              if (!this.hls.levels[this.hls.loadLevel]?.details) {
+              // Do not seek when selected variant playlist is unloaded
+              if (!this.hls.loadLevelObj?.details) {
                 return 0;
               }
-              // Do not seek when loading frags
+              // Do not seek when required fragments are inflight or appending
               const inFlightDependency = getInFlightDependency(
                 this.hls.inFlightFragments,
                 startTime,

--- a/src/controller/gap-controller.ts
+++ b/src/controller/gap-controller.ts
@@ -12,7 +12,7 @@ import type { InFlightData } from './base-stream-controller';
 import type { InFlightFragments } from '../hls';
 import type Hls from '../hls';
 import type { FragmentTracker } from './fragment-tracker';
-import type { Fragment } from '../loader/fragment';
+import type { Fragment, MediaFragment } from '../loader/fragment';
 import type { SourceBufferName } from '../types/buffer';
 import type {
   BufferAppendedData,
@@ -503,7 +503,7 @@ export default class GapController extends TaskLoop {
    * @param partial - The partial fragment found at the current time (where playback is stalling).
    * @private
    */
-  private _trySkipBufferHole(partial: Fragment | null): number {
+  private _trySkipBufferHole(partial: MediaFragment | null): number {
     const { fragmentTracker, media } = this;
     const config = this.hls?.config;
     if (!media || !fragmentTracker || !config) {
@@ -515,7 +515,7 @@ export default class GapController extends TaskLoop {
     const bufferInfo = BufferHelper.bufferInfo(media, currentTime, 0);
     const startTime =
       currentTime < bufferInfo.start ? bufferInfo.start : bufferInfo.nextStart;
-    if (startTime) {
+    if (startTime && this.hls) {
       const bufferStarved = bufferInfo.len <= config.maxBufferHole;
       const waiting =
         bufferInfo.len > 0 && bufferInfo.len < 1 && media.readyState < 3;
@@ -541,6 +541,19 @@ export default class GapController extends TaskLoop {
                 PlaylistLevelType.MAIN,
               );
             if (startProvisioned) {
+              // Do not seek when switching variants
+              if (!this.hls.levels[this.hls.loadLevel]?.details) {
+                return 0;
+              }
+              // Do not seek when loading frags
+              const inFlightDependency = getInFlightDependency(
+                this.hls.inFlightFragments,
+                startTime,
+              );
+              if (inFlightDependency) {
+                return 0;
+              }
+              // Do not seek if we can't walk tracked fragments to end of gap
               let moreToLoad = false;
               let pos = startProvisioned.end;
               while (pos < startTime) {
@@ -567,7 +580,7 @@ export default class GapController extends TaskLoop {
         );
         this.moved = true;
         media.currentTime = targetTime;
-        if (!partial?.gap && this.hls) {
+        if (!partial?.gap) {
           const error = new Error(
             `fragment loaded with buffer holes, seeking from ${currentTime} to ${targetTime}`,
           );

--- a/src/controller/interstitials-controller.ts
+++ b/src/controller/interstitials-controller.ts
@@ -1921,7 +1921,7 @@ Schedule: ${scheduleItems.map((seg) => segmentToString(seg))}`,
     const userConfig = primary.userConfig;
     let videoPreference = userConfig.videoPreference;
     const currentLevel =
-      primary.levels[primary.loadLevel] || primary.levels[primary.currentLevel];
+      primary.loadLevelObj || primary.levels[primary.currentLevel];
     if (videoPreference || currentLevel) {
       videoPreference = Object.assign({}, videoPreference);
       if (currentLevel.videoCodec) {

--- a/src/controller/level-controller.ts
+++ b/src/controller/level-controller.ts
@@ -389,6 +389,10 @@ export default class LevelController extends BasePlaylistController {
     return this._levels;
   }
 
+  get loadLevelObj(): Level | null {
+    return this.currentLevel;
+  }
+
   get level(): number {
     return this.currentLevelIndex;
   }

--- a/src/controller/stream-controller.ts
+++ b/src/controller/stream-controller.ts
@@ -237,7 +237,7 @@ export default class StreamController
 
   protected onTickEnd() {
     super.onTickEnd();
-    if (this.media?.readyState) {
+    if (this.media?.readyState && this.media.seeking === false) {
       this.lastCurrentTime = this.media.currentTime;
     }
     this.checkFragmentChanged();

--- a/src/hls.ts
+++ b/src/hls.ts
@@ -690,8 +690,18 @@ export default class Hls implements HlsEventEmitter {
     return levels ? levels : [];
   }
 
+  /**
+   * @returns LevelDetails of last loaded level (variant) or `null` prior to loading a media playlist.
+   */
   get latestLevelDetails(): LevelDetails | null {
     return this.streamController.getLevelDetails() || null;
+  }
+
+  /**
+   * @returns Level object of selected level (variant) or `null` prior to selecting a level or once the level is removed.
+   */
+  get loadLevelObj(): Level | null {
+    return this.levelController.loadLevelObj;
   }
 
   /**

--- a/src/utils/rendition-helper.ts
+++ b/src/utils/rendition-helper.ts
@@ -502,5 +502,5 @@ export function useAlternateAudio(
   audioTrackUrl: string | undefined,
   hls: Hls,
 ): boolean {
-  return !!audioTrackUrl && audioTrackUrl !== hls.levels[hls.loadLevel]?.uri;
+  return !!audioTrackUrl && audioTrackUrl !== hls.loadLevelObj?.uri;
 }

--- a/tests/unit/controller/gap-controller.ts
+++ b/tests/unit/controller/gap-controller.ts
@@ -16,7 +16,7 @@ import {
 import { MockMediaElement, MockMediaSource } from '../utils/mock-media';
 import type { HlsConfig } from '../../../src/config';
 import type StreamController from '../../../src/controller/stream-controller';
-import type { Fragment } from '../../../src/loader/fragment';
+import type { Fragment, MediaFragment } from '../../../src/loader/fragment';
 
 chai.use(sinonChai);
 const expect = chai.expect;
@@ -202,7 +202,7 @@ describe('GapController', function () {
       const bufferInfo = BufferHelper.bufferedInfo([], 0, 0);
       sandbox
         .stub(gapController.fragmentTracker, 'getPartialFragment')
-        .returns({} as unknown as Fragment);
+        .returns({} as unknown as MediaFragment);
       const skipHoleStub = sandbox.stub(gapController, '_trySkipBufferHole');
       gapController._tryFixBufferStall(bufferInfo, 100);
       expect(skipHoleStub).to.have.been.calledOnce;

